In [1]:
%pip install git+https://github.com/awslabs/fast-differential-privacy

  Cloning https://github.com/awslabs/fast-differential-privacy to /tmp/pip-req-build-n2wccy4a
  Running command git clone --filter=blob:none --quiet https://github.com/awslabs/fast-differential-privacy /tmp/pip-req-build-n2wccy4a
  Resolved https://github.com/awslabs/fast-differential-privacy to commit 1202baa441c38c338415bc24e4cf12514c46bac0
  Preparing metadata (setup.py) ... done
  Created wheel for fastDP: filename=fastDP-1.1.0-py3-none-any.whl size=101664 sha256=cf1accc3de7649cf3b54276f4dca95602df77255c0bfc4cec9c3fb084915364f
  Stored in directory: /tmp/pip-ephem-wheel-cache-e8612w00/wheels/b3/2b/b5/62668a08336b29e28ca790358738474aa8c2c68359b494f454
Successfully built fastDP


In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
%cd /content/drive/Othercomputers/My Mac/NormaLcifar10/

/content/drive/Othercomputers/My Mac/NormaLcifar10


In [ ]:
!python finetuning.py --name fttrial --seed 0 --epoch 50 --epsilon 1 --b 100 --lr 1 --gamma 0.8

Files already downloaded and verified
Files already downloaded and verified
Downloading: "https://download.pytorch.org/models/resnet18-f37072fd.pth" to /root/.cache/torch/hub/checkpoints/resnet18-f37072fd.pth
100% 44.7M/44.7M [00:00<00:00, 67.5MB/s]
Number of trainable components:  22 ; Number of trainable layers:  21
>>>>>>>>>>>>>>>>> Applying  automatic  per-sample gradient clipping.
>>>>>>>>>>>>>>>>> Block heads for per-sample gradient clipping are defined as: ['conv1']
noise multiplier 1.4625244140625
/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py:1359: UserWarning: Using a non-full backward hook when the forward contains multiple autograd Nodes is deprecated and will be removed in future versions. This hook will be missing some grad_input. Please use register_full_backward_hook to get the documented behavior.
  warnings.warn("Using a non-full backward hook when the forward contains multiple autograd Nodes "
['bn1.weight', 'bn1.bias', 'layer1.0.bn1.weight', 'lay

- !python finetuning.py --name fttrial --seed 0 --epoch 50 --epsilon 1.5 --b 100 --lr 0.01

  0 %,1 %,4 %,7 %,13 %,21 %,28 %,35 %,40 %,44 %,47 %,48 %,49 %,49 %,47 %,44 %,43 %

  noise multiplier 1.1159423828125

- !python finetuning.py --name fttrial --seed 0 --epoch 50 --epsilon 1 --b 100 --lr 0.01

  noise multiplier 1.4625244140625